packages/plugin-ext/src/plugin/in-plugin-filesystem-proxy.ts

```diff
@@ -0,0 +1,61 @@
+/********************************************************************************
+ * Copyright (C) 2020 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { TextEncoder, TextDecoder } from 'util';
+import { FileSystemMain } from '../common/plugin-api-rpc';
+import { UriComponents } from '../common/uri-components';
+import { FileSystem, FileSystemError } from './types-impl';
+
+/**
+ * This class is managing FileSystem proxy
+ */
+export class InPluginFileSystemProxy implements FileSystem {
+
+    private proxy: FileSystemMain;
+
+    constructor(proxy: FileSystemMain) {
+        this.proxy = proxy;
+    }
+
+    async readFile(uri: UriComponents): Promise<Uint8Array> {
+        try {
+            const val = await this.proxy.$readFile(uri);
+            return new TextEncoder().encode(val);
```

Comment:
I'm worried about this line: will it provide the exact byte contents of the file in every case? Reading the file as a string will use a certain text encoding. How can we be sure we're using the same encoding on this line?

Reply:
@tsmaeder because it is using utf-8 ?

Reply:
But you don't know that: resources are a point of extension. Say a resource reads a file in EBDIC and translates that correctly to a string. When you encode that using utf-8, you will not have the original byte contents in your byte array.

Reply:
yes, it was like temporary issue (for non utf-8 resources) until being able to use filesystem from `workspace.fs` (and then hook FileStat etc in theia core)
For now we only have `resources` allowing the file access + custom FileSystemResourceProvider